sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureBasicFileAttributes.java

```diff
@@ -136,17 +144,17 @@ public boolean isSymbolicLink() {
      */
     @Override
     public boolean isOther() {
-        return false;
+        return this.internalAttributes.isOther();
     }
 
     /**
-     * Returns the size of the file (in bytes).
+     * Returns the size of the file (in bytes) or null if this is a virtual directory.
      *
-     * @return the size of the file
+     * @return the size of the file or null if this is a virtual directory
      */
     @Override
     public long size() {
-        return properties.getBlobSize();
+        return this.internalAttributes.size();
     }
```

Comment:
You are returning little `l` `long` here - but your comment says it returns null if a virtual directory?

Reply:
Good catch! Thank you. I think in the case of regular directories we return 0, and there's probably no need to distinguish here, so I'll change this to return 0 and update the comment.